cmd/nerdctl/login_linux_test.go

```diff
@@ -57,269 +52,485 @@ func (ag *Client) WithHostsDir(hostDirs string) *Client {
 }
 
 func (ag *Client) WithCredentials(username, password string) *Client {
-	ag.args = append(ag.args, "--username", username, "--password", password)
+	if username != "" {
+		ag.args = append(ag.args, "--username", username)
+	}
+	if password != "" {
+		ag.args = append(ag.args, "--password", password)
+	}
 	return ag
 }
 
+func (ag *Client) WithConfigPath(value string) *Client {
+	ag.configPath = value
+	return ag
+}
+
+func (ag *Client) GetConfigPath() string {
+	return ag.configPath
+}
+
 func (ag *Client) Run(base *testutil.Base, host string) *testutil.Cmd {
 	if ag.configPath == "" {
 		ag.configPath, _ = os.MkdirTemp(base.T.TempDir(), "docker-config")
 	}
-	args := append([]string{"--debug-full", "login"}, ag.args...)
+	args := []string{"login"}
+	if base.Target == "nerdctl" {
+		args = append(args, "--debug-full")
+	}
+	args = append(args, ag.args...)
 	icmdCmd := icmd.Command(base.Binary, append(base.Args, append(args, host)...)...)
 	icmdCmd.Env = append(base.Env, "HOME="+os.Getenv("HOME"), "DOCKER_CONFIG="+ag.configPath)
+
 	return &testutil.Cmd{
 		Cmd:  icmdCmd,
 		Base: base,
 	}
 }
 
-func TestLogin(t *testing.T) {
+func TestLoginPersistence(t *testing.T) {
+	base := testutil.NewBase(t)
+	t.Parallel()
+
+	// Retrieve from the store
+	testCases := []struct {
+		auth string
+	}{
+		{
+			"basic",
+		},
+		{
+			"token",
+		},
+	}
+
+	for _, tc := range testCases {
+		t.Run(fmt.Sprintf("Server %s", tc.auth), func(t *testing.T) {
```

Comment:
I think you need to add a `tc := tc` between 107 and 108 to make sure different tests use separated tc.